download-directory/Jenkinsfile

```diff
@@ -0,0 +1,65 @@
+import groovy.json.JsonSlurper
+
+pipeline {
+    	agent any
+
+	stages {
+		stage('Check upstream builds succeeded'){
+			steps{
+				script{
+					def addLinksInsertionStatusUrl = httpRequest authentication: 'jenkinsKey', url: "${env.JENKINS_JOB_URL}/job/${env.RELEASE_NUMBER}/job/AddLinks-Insertion/lastBuild/api/json"
+					def addLinksInsertionJson = new JsonSlurper().parseText(addLinksInsertionUrl.getContent())
+					if(addLinksInsertionJson['result'] != "SUCCESS"){
+						error("Most recent AddLinks-Insertion build status: " + addLinksInsertionJson['result'] + ". Please complete a successful build.")
+					}
+			    	}
+			}
+		}
+	    
+		stage('Setup: Install Pathway Exchange artifact'){
+			steps{
+				script{
+					dir('download-directory'){
+						sh "./build_pathway_exchange.sh"
+					}
+				}
+			}
+		}
+        	stage('Setup: Build DownloadDirectory archive'){
+            		steps{
+				script{
+                    			dir('download-directory'){
+						sh "mvn clean package -DskipTests"
+			    			sh "rm -rf downloadDirectory"
+						sh "unzip -o target/downloadDirectory-distr.zip"
+                			}
+          			}
+            		}
+        	}
+		stage('Main: Run DownloadDirectory'){
+			steps{
+				script{
+					dir('download-directory'){
+						withCredentials([file(credentialsId: 'Config', variable: 'ConfigFile')]){
+							withCredentials([file(credentialsId: 'stepsToRun', variable: 'StepsToRun')]){
```

Comment:
How does `StepsToRun` get configured?

Reply:
In Jenkins, I have it so that it is stored as a secret file. This way if you still want to run specific steps in download directory, its still possible. All you have to do is upload the file with the ones you want to skip commented out

Reply:
Ok. If there's a README for the Jenkins usage, this should be in it.